sim/core/buffs.go

```diff
@@ -755,6 +759,52 @@ func UnholyFrenzyAura(character *Character, actionTag int32) *Aura {
 	})
 }
 
+func registerRevitalizeHotCD(agent Agent, label string, hotID ActionID, ticks int, tickPeriod time.Duration, uptimeCount int32) {
+	if uptimeCount == 0 {
+		return
+	}
+
+	character := agent.GetCharacter()
+	revActionID := ActionID{SpellID: 48545}
+
+	manaMetrics := character.NewManaMetrics(revActionID)
+	energyMetrics := character.NewEnergyMetrics(revActionID)
+	rageMetrics := character.NewRageMetrics(revActionID)
+	rpMetrics := character.NewRunicPowerMetrics(revActionID)
+
+	// Calculate desired downtime based on selected uptimeCount (1 count = 10% uptime, 0%-100%)
+	totalDuration := time.Duration(ticks) * tickPeriod
+	uptimePercent := float64(uptimeCount) / 100.0
+
+	aura := character.GetOrRegisterAura(Aura{
+		Label:    "Revitalize-" + label,
+		ActionID: hotID,
+		Duration: totalDuration,
+		OnGain: func(aura *Aura, sim *Simulation) {
+			pa := NewPeriodicAction(sim, PeriodicActionOptions{
+				Period:   tickPeriod,
+				NumTicks: ticks,
+				OnAction: func(s *Simulation) {
+					if s.RandomFloat("Revitalize Proc") < 0.15 {
+						if aura.Unit.HasManaBar() {
+							aura.Unit.AddMana(s, aura.Unit.BaseMana*0.01, manaMetrics, true)
```

Comment:
Are we sure this is 1% base mana and not 1% max mana? The latter seems more likely.

Reply:
4b65f50e86602d9d83103bf815d75845ceaa7329